src/mlpack/methods/logistic_regression/logistic_regression_impl.hpp

```diff
@@ -98,17 +139,50 @@ double LogisticRegression<MatType>::Train(
   return out;
 }
 
+template<typename MatType>
+template<typename OptimizerType, typename... CallbackTypes, typename, typename>
+typename LogisticRegression<MatType>::ElemType
+LogisticRegression<MatType>::Train(
+    const MatType& predictors,
+    const arma::Row<size_t>& responses,
+    OptimizerType& optimizer,
+    const double lambda,
+    CallbackTypes&&... callbacks)
+{
+  this->lambda = lambda;
+  return Train(predictors, responses, optimizer,
+      std::forward<CallbackTypes>(callbacks)...);
+}
+
 template<typename MatType>
 template<typename VecType>
 size_t LogisticRegression<MatType>::Classify(const VecType& point,
                                              const double decisionBoundary)
     const
 {
   return size_t(1.0 / (1.0 + std::exp(-parameters(0) - arma::dot(point,
-      parameters.tail_cols(parameters.n_elem - 1)))) +
+      parameters.tail_cols(parameters.n_elem - 1).t()))) +
       (1.0 - decisionBoundary));
 }
 
+template<typename MatType>
+template<typename VecType>
```

Comment:
The issue here is, what if I do `lr.Classify(matrix.col(i))`?  Then `MatType` might be `arma::mat`, but `VecType` will be `arma::subview_col` (or similar).  So I have kept the types separate to allow this flexibility... if we made the first parameter `MatType` instead of `VecType`, then calling `lr.Classify(matrix.col(i))` would compile, but the `matrix.col(i)` subexpression would be formed into its own `arma::mat` via copy.